pkg/controller/statusmanager/pod_status.go

```diff
@@ -276,3 +286,7 @@ func (status *StatusManager) setLastPodState(
 		return status.client.Patch(context.TODO(), newStatus, client.MergeFrom(oldStatus))
 	})
 }
+
+func isNonCritical(obj metav1.Object) bool {
+	return obj.GetAnnotations()[names.NonCriticalAnnotation] == "true"
```

Comment:
Since this condition is a boolean, the convention would be to set an empty string for true.